core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt

```diff
@@ -0,0 +1,372 @@
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog.OpenCredits
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog.SelectFolder
+import org.kiwix.kiwixmobile.core.utils.files.FileUtils.getPathFromUri
+import java.io.File
+import java.util.Locale
+import javax.inject.Inject
+
+abstract class CorePrefsFragment :
+  PreferenceFragmentCompat(),
+  SettingsContract.View,
+  OnSharedPreferenceChangeListener {
+  @JvmField @Inject
+  internal var presenter: SettingsPresenter? = null
+
+  @JvmField @Inject
+  var sharedPreferenceUtil: SharedPreferenceUtil? = null
+
+  @JvmField @Inject
+  var storageCalculator: StorageCalculator? = null
+
+  @JvmField @Inject
+  var nightModeConfig: NightModeConfig? = null
+
+  @JvmField @Inject
+  var alertDialogShower: DialogShower? = null
+  override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+    coreComponent
+      .activityComponentBuilder()
+      .activity(requireActivity())
+      .build()
+      .inject(this)
+    addPreferencesFromResource(R.xml.preferences)
+    setStorage()
+    setUpSettings()
+    setupZoom()
+    sharedPreferenceUtil?.let {
+      LanguageUtils(requireActivity()).changeFont(
+        requireActivity().layoutInflater,
+        it
+      )
+    }
+  }
+
+  private fun setupZoom() {
+    val textZoom = findPreference<Preference>(INTERNAL_TEXT_ZOOM)
+    textZoom?.onPreferenceChangeListener =
+      Preference.OnPreferenceChangeListener { _: Preference?, newValue: Any ->
+        sharedPreferenceUtil?.textZoom = (newValue as Int + ZOOM_OFFSET) * ZOOM_SCALE
+        updateTextZoomSummary(textZoom)
+        true
+      }
+    updateTextZoomSummary(textZoom)
+  }
+
+  private fun updateTextZoomSummary(textZoom: Preference?) {
+    textZoom?.summary = getString(R.string.percentage, sharedPreferenceUtil!!.textZoom)
+  }
+
+  protected abstract fun setStorage()
+  override fun onResume() {
+    super.onResume()
+    preferenceScreen.sharedPreferences
+      .registerOnSharedPreferenceChangeListener(this)
+  }
+
+  override fun onPause() {
+    super.onPause()
+    preferenceScreen.sharedPreferences
+      .unregisterOnSharedPreferenceChangeListener(this)
+  }
+
+  private fun setUpSettings() {
+    setAppVersionNumber()
+  }
+
+  protected fun setUpLanguageChooser(preferenceId: String) {
+    findPreference<ListPreference>(
+      preferenceId
+    )?.let {
+      var languageCodeList = LanguageUtils(requireActivity()).keys
+      languageCodeList = listOf(Locale.ROOT.language) + languageCodeList
+      sharedPreferenceUtil?.let { sharedPreferenceUtil ->
+        val selectedLang = selectedLanguage(languageCodeList, sharedPreferenceUtil.prefLanguage)
+        it.entries = languageDisplayValues(languageCodeList)
+        it.entryValues = languageCodeList.toTypedArray()
+        it.setDefaultValue(selectedLang)
+        it.value = selectedLang
+        it.title =
```

Comment:
Maybe we should replace the lambda parameter `it` with `languagePref` for the sake of readabilty.

Reply:
done